src/05-derivatives/04-functional-derivatives.md

```diff
@@ -0,0 +1,373 @@
+
+Regressors will generally be stored as tables, with a row of column headers
+indicating the name of the regressor. In the case where every voxel has a
+regressor, then the data should be stored in a 4D NIfTI file.
+
+Tabular regressor files MUST be accompanied by a data dictionary in JSON format,
+consistent with the format described in
+[Common princioples](../02-common-principles.md).
+
+Volumetric regressor files MAY be accompanied by a data dictionary in JSON
+format, with OPTIONAL `LongName`, `Description`, `Levels`, `Units`, and
+`TermURL` fields.
+
+### General time series and regressors
+
+Template:
+
+```Text
+<pipeline_name>/
+    sub-<participant_label>/
+        func/
+	        <source_keywords>[_desc-<label>]_regressors.tsv
+	        <source_keywords>[_desc-<label>]_regressors.json
+```
+
+for example:
+
+```Text
+sub-001/
+    func/
+        sub-001_task-rest_run-1_desc-confounds_regressors.tsv
+        sub-001_task-rest_run-1_desc-confounds_regressors.json
+```
+
+Any time series or regressor with one value per BOLD volume may be stored in a
+regressors file. If a column is specified in another sub-section, then any
+additional required metadata MUST be stored in the JSON description of the
+column.
+
+#### Column names
+
+Column names are unique alphanumeric values that are defined in a relevant JSON
+sidecar file. For a series of related columns, it is RECOMMENDED to use
+zero-based indexing as a suffix. For example, the first six aCompCor components
+may be named `a_comp_cor_00` ... `a_comp_cor_05`. Custom column names must not
+conflict with reserved names specified in this document.
+
+Reserved names are specified in the remaining subsections.
+
+#### Time series transformations
+
+Any column may have the following suffixes appended to indicate transformations
+applied to the original time series data:
+
+| Transformation | Description                                                                 |
+| -------------- | --------------------------------------------------------------------------- |
+| `_shift_back`  | The time series has been lagged by one TR. `x_shift_back[i] = x[i-1]`       |
+| `_dt`          | The discrete first derivative of the time series. `x_dt[i] = x[i+1] - x[i]` |
+| `_sq`          | The square of the time series                                               |
+| `_var_norm`    | The time series has been variance normalized                                |
+| `_centered`    | The time series has had its mean subtracted                                 |
+
+For example, `rot_z_shift_back_sq` means the square of the lagged version of the
+Z rotation (see Motion-related regressors).
+
+### ROI-based time series extraction
+
+Template:
+
+```Text
+<pipeline_name>/
+    sub-<participant_label>/
+        func/
+            <source_keywords>[_atlas-<atlas_label>][_desc-<label>]_timeseries.<tsv|nii.gz>
+            <source_keywords>[_atlas-<atlas_label>][_desc-<label>]_timeseries.json
```

Comment:
I guess one question then is what to call these files if you want your ROI time series to be segregated from your confounds. `atlas-` doesn't apply to generic time series. `_roiseries.tsv`?

Reply:
Sorry, is the concern that we can't use the `atlas` key-value pair with `_timeseries.tsv` ? I'm not yet sure I understand why we'd need another suffix....

Reply:
The concern is that `atlas-` doesn't sensibly apply to all `_timeseries.tsv` files, but it does for ROI time series. There's a semantic minefield ahead, so I'm going to plant some flags, which are:

| Concept | Old suffix | New suffix |
|-|-|-|
| Generic 1/TR time series | `_regressors` | `_timeseries`(b) |
| ROI-based time series | `_timeseries`(a) | `_???` |

Hopefully that is at least slightly clarifying. Onward.

I'm realizing now that `atlas-` is the only non-`desc-` keyword that we've been allowing in these `_regressors` subsets, so it may be the case that we simply want to drop the `atlas-` keyword in `_timeseries`(b).

But to expand my thinking, we are considering in these sections various subsets of `_regressors`/`_timeseries`(b), and each subset could potentially have had valid keywords that didn't apply to all of the others. If we want to keep the `atlas-` keyword, then this subset of ROI-based time series (`_timeseries`(a)/`_???`) cannot simply disappear, as `atlas-` does not apply to all `_timeseries`(b).

So if we want `_???` to be an actual suffix indicating ROI time series and having the `atlas-` flag, we need to decide what `_???` should be.

😟  Was that clear?

Reply:
yeah I think this is something that comes up in the anat derivatives too. does it make sense to have an "atlas" keyword? I see it elsewhere as "label"... would be good to maybe clarify in the common intro? Another choice is dseg? (instead of atlas)  - because it should map to a _dseg file?

Reply:
Ah, ok ! I think it was clear, but I suppose we'll find out from my next question :laughing: 

If I understand correctly, deriving a new suffix seems to just be reframing the problem. The original concern with `_regressors.tsv` vs `_timeseries.tsv` is that they encode the same kind of information, just with a different intent. By collapsing these only some `_timeseries.tsv` could (reasonably) use the `atlas` key-value pair.

If that's all true, then I think the concern would be if we wanted to **require** the `atlas` key-value pair _but only for_ ROI time series. From this current draft it seems that `atlas` as a key-value pair is optional, so I don't see the problem with keeping just `_timeseries.tsv`.

Does that roughly match your understanding, @effigies ??

Reply:
Yes, sorry I missed your response @edickie ! Option 2 is to just remove that keyword altogether.

Reply:
@emdupre Yes, `atlas` is optional, and I think was mainly intended for cases where you expect to create time series from several atlases and need a meaningful discriminant. I think we could say "it's only valid in these cases", it'll just be harder to write a validation rule. The `Atlas` sidecar metadata is still required for any ROI-based time series.

@edickie I don't think we can use `label-`, since that seems to indicate a single, specific mask (like `BG`) in the `anat/` world, while `atlas-` in principle refers to any segmentation/parcellation (with masks being legal, as they can be treated as a degenerate segmentation). I don't have strong feelings on `dseg-`, but do we have other cases of suffix/keyword overlap?

Reply:
Was this resolved by https://github.com/chrisfilo/bids-specification/pull/2?